### DISCUSSION TITLE
help set-up trigger.dev on stackblitz

### DISCUSSION THREAD

**Discussion by EdHossam** [triggerdotdev/trigger.dev#1462] [Q&A]
Hi,

been fighting with this for a while now, and feel that I am going nowhere, maybe I am missing something fundamental!.

the issue is I have a small project using Nextjs on [https://stackblitz.com/](url) , and I am trying to setup trigger.dev on it also to have a quick play with the functionality and tasks, currently every time I try to run any of the CLI commands to set-up trigger. dev like "npx trigger.dev@latest init" or "npx trigger.dev@latest dev" it always comes back with the error:
"Failed to create authorization code
Connection error.
X Error: You must login first. Use the `login` CLI command."

no matter what I tried or what CLI commands I added ( like adding --profile to the login CLI command) , it can't generate the authorization code!

I went through all the steps in the guides and docs, and set up my env. file, the trigger.config.ts file, and had a task written in typescript inside the "trigger" folder as specified by the default [dirs] config in the trigger.config.ts file.

if I tried to trigger the task , it always comes back with the error "ApiClientMissingError: You need to set the TRIGGER_SECRET_KEY environment", which I already did in the .env file, and also added .env.local file !!

have anyone been able to successfully get trigger.dev working on an online environment like stackblitz or replit ?

Appreciate your help on this!

Thanks,
Ed

**Comment by bvergara87** [comment]
I am also having the same issue when trying to deploy via Github Actions.

**Comment by blessingefkt** [comment]
this is still an issue for me as well, but I'm on my own ubuntu box
